src/main/java/uk/gov/hmcts/darts/cases/helper/AdvancedSearchRequestHelper.java

```diff
@@ -190,8 +224,12 @@ private Join<HearingEntity, CourtroomEntity> joinCourtroom(Root<CourtCaseEntity>
 
     }
 
+    @SuppressWarnings("unchecked")
     private Join<CourtCaseEntity, CourthouseEntity> joinCourthouse(Root<CourtCaseEntity> caseRoot) {
-        return caseRoot.join(CourtroomEntity_.COURTHOUSE, JoinType.INNER);
+        Optional<Join<CourtCaseEntity, ?>> foundJoin = caseRoot.getJoins().stream().filter(join -> join.getAttribute().getName().equals(
```

Comment:
Why do we need to do all this - looking at Table "darts.court_case" - we would always expect to have this relationship?
`cth_id           | integer                  |           | not null `